File: Navtrack.Core/ClientApp/src/services/api/types/location/LocationModel.ts
export type LocationModel = {
  id: number,
  assetId: number,
  latitude: number,
  longitude: number,
  dateTime: Date,
  speed: number,
  heading: number,
  altitude: number,
  satellites: number,
  hdop: number
};

export const DefaultLocationModel : LocationModel = {
  id: 0,
  assetId: 0,
  latitude: 0,
  longitude: 0,
  dateTime: new Date(),
  speed: 0,
  heading: 0,
  altitude: 0,
  satellites: 0,
  hdop: 0
};

File: backend/Navtrack.Api.Model/Locations/LocationModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Navtrack.Api.Model.Locations;

public class LocationModel
{
    public string Id { get; set; }

    [Required]
    public double Longitude => Coordinates[0];
    
    [Required]
    public double Latitude => Coordinates[1];


    [Required]
    public double[] Coordinates { get; set; }

    [Required]
    public bool ValidCoordinates => CoordinatesValidation.IsValidLatitude(Latitude) &&
                                    CoordinatesValidation.IsValidLongitude(Longitude);

    [Required]
    public DateTime DateTime { get; set; }

    public float? Speed { get; set; }
    public float? Heading { get; set; }
    public float? Altitude { get; set; }
    public int? Satellites { get; set; }
    public float? HDOP { get; set; }
    public bool? Valid { get; set; }
    public short? GsmSignal { get; set; }
    public double? Odometer { get; set; }
}